lib/routes/mangadex/user/follows.ts

```diff
@@ -0,0 +1,154 @@
+It's recommended to use the \`/mangadex/mdlist/:listId?\` route instead for better performance, though it requires manual configuration.`,
+    categories: ['anime'],
+    parameters: {
+        type: {
+            description: 'The type of follows to fetch',
+            default: 'reading',
+            options: [
+                { value: 'reading', label: 'Reading' },
+                { value: 'plan-to-read', label: 'Plan to Read' },
+                { value: 'completed', label: 'Completed' },
+                { value: 'on-hold', label: 'On Hold' },
+                { value: 're-reading', label: 'Re-reading' },
+                { value: 'dropped', label: 'Dropped' },
+            ],
+        },
+    },
+    radar: [
+        {
+            source: ['mangadex.org/titles/follows'],
+            target: '/user/follow/reading',
+        },
+    ],
+    features: {
+        requireConfig: [
+            {
+                name: 'MANGADEX_USERNAME',
+                description: 'MangaDex Username, required when refresh-token is not set',
+                optional: true,
+            },
+            {
+                name: 'MANGADEX_PASSWORD',
+                description: 'MangaDex Password, required when refresh-token is not set',
+                optional: true,
+            },
+            {
+                name: 'MANGADEX_CLIENT_ID',
+                description: 'MangaDex Client ID',
+                optional: false,
+            },
+            {
+                name: 'MANGADEX_CLIENT_SECRET',
+                description: 'MangaDex Client Secret',
+                optional: false,
+            },
+            {
+                name: 'MANGADEX_REFRESH_TOKEN',
+                description: 'MangaDex Refresh Token, required when username and password are not set',
+                optional: true,
+            },
+            {
+                name: 'MANGADEX_FILTERED_LANGUAGES',
+                description: 'MangaDex Filtered Languages',
+                optional: true,
+            },
+        ],
+    },
+    handler,
+};
+
+async function handler(ctx) {
+    const userFollowUrl = 'https://api.mangadex.org/manga/status';
+
+    const { type } = ctx.req.param();
+
+    const followType = (type || 'reading') as FollowType;
+
+    const accessToken = await getToken();
+
+    const statuses = (await cache.tryGet(
+        `mangadex:user-follow-${followType}`,
+        async () => {
+            const response = await got.get(userFollowUrl, {
+                headers: {
+                    Authorization: `Bearer ${accessToken}`,
+                    'User-Agent': config.trueUA,
+                },
+            });
+
+            const statuses = response?.data?.statuses;
+            if (!statuses) {
+                throw new Error('Failed to retrieve user follows from MangaDex API.');
+            }
+
+            return statuses;
+        },
+        config.cache.contentExpire,
```

Comment:
https://github.com/DIYgod/RSSHub/pull/18269#discussion_r2072440789
```suggestion
        config.cache.routeExpire,
```